File: headless_solutions/1_bit_and_2_bit_characters/solution.cpp
// C++ implementation matching signature
class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        int i = 0;
        while (i < (int)bits.size() - 1) {
            if (bits[i] == 0) {
                ++i;          // 1‑bit character
            } else {
                i += 2;       // 2‑bit character
            }
        }
        return i == (int)bits.size() - 1;
    }
};

File: headless_solutions/1_bit_and_2_bit_characters/solution.py
import collections
import functools
import itertools
import math
import heapq
import bisect
from typing import *
from collections import *
from functools import *
from heapq import *
from bisect import *

# Python 3 implementation matching signature
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        i = 0
        while i < len(bits) - 1:
            if bits[i] == 0:
                i += 1          # 1‑bit character
            else:
                i += 2          # 2‑bit character
        return i == len(bits) - 1
